File: app/src/main/java/org/odddev/fantlab/core/di/Injector.java
package org.odddev.fantlab.core.di;

import android.support.annotation.NonNull;

/**
 * @author kenrube
 * @since 23.08.16
 */

public class Injector {

    private static AppComponent appComponent;

    public static void init(@NonNull AppComponent appComponent) {
        Injector.appComponent = appComponent;
    }

    @NonNull
    public static AppComponent getAppComponent() {
        if (appComponent == null) throw new RuntimeException("AppComponent not initialized yet!");
        return appComponent;
    }
}

File: app/src/main/java/org/odddev/fantlab/core/di/Injector.kt
package org.odddev.fantlab.core.di

/**
 * @author kenrube
 * *
 * @since 23.08.16
 */

object Injector {

	private var appComponent: AppComponent? = null

	fun init(appComponent: AppComponent) {
		Injector.appComponent = appComponent
	}

	fun getAppComponent(): AppComponent {
		if (appComponent == null) throw RuntimeException("AppComponent not initialized yet!")
		return appComponent as AppComponent
	}
}
